### ISSUE TITLE
New function "stream_file" about "append" flag

### ISSUE THREAD

**Issue by hdcscy** [chriskohlhoff/asio#1144]
I have a simple demo here on windows:
`void async_append(const std::string& data,
                    const std::function<void(const boost::system::error_code&,
                                             size_t)>& handler)
  {
    try {
      boost::asio::stream_file file(
        m_ctx, *m_path, boost::asio::stream_file::append);
      if (file.is_open()) {
        file.async_write_some(
          boost::asio::const_buffers_1(data.data(), data.size()), handler);
        file.close();
        if (m_ctx.stopped()) {
          m_ctx.restart();
        }
        m_ctx.run();
      }
    } catch (boost::exception& ex) {
      // TODO:
    }
  }`

As the title, the **system:5** error is returned in callback, which is equivalent to **ERROR_ACCESS_DENIED**.
I've tried combining with other flags, but either throws an exception or doesn't work.
However, e.g. **write_only** and **read_only**, they are valid and do not generate errors.

**Comment by klemens-morgenstern** [comment]
Do you have access rights dor the file?

**Comment by hdcscy** [comment]
> Do you have access rights dor the file?

Hello,
Yes, I'm administrator, and UAC permission is disabled on Windows.
I have tried STL's **fstream** as well as C standard library's **FILE** and both nice work for **append**.

**Comment by klemens-morgenstern** [comment]
Aren't append & truncate mutually exclusive?

**Comment by hdcscy** [comment]
> Aren't append & truncate mutually exclusive?

Yes, they are mutually exclusive, will throw an exception. but apend and other flag bit combinations also do not work, including its own use alone.
So I tried many times, no solution.

**Comment by klemens-morgenstern** [comment]
write_only + append throws?

**Comment by hdcscy** [comment]
> write_only + append throws?

Although this combination will not throw an exception, but will not append, only overwrite, the **append** flag does not work.
